packages/react/xr/src/controller.tsx

```diff
@@ -87,3 +101,72 @@ export const XRControllerModel = forwardRef<Object3D, XRControllerModelOptions>(
   useFrame(update)
   return <primitive object={model} />
 })
+
+export function useLoadXRControllerLayout(
+  profileIds: string[],
+  handedness: XRHandedness,
+  options?: XRControllerLayoutLoaderOptions,
+): XRControllerLayout | undefined {
+  const loader = useRef(new XRControllerLayoutLoader(options))
+
+  const [layout, setLayout] = useState<XRControllerLayout | undefined>(undefined)
+
+  useEffect(() => {
+    const load = async () => {
+      setLayout(await loader.current.load(profileIds, handedness))
+    }
+    load()
+  }, [handedness, profileIds])
+
+  return layout
+}
+
+export function useLoadXRControllerModel(layout: XRControllerLayout) {
+  return suspend(
+    (layout, loader, qwre) => (layout ? loadXRControllerModel(layout, loader) : Promise.resolve()),
+    [layout, undefined, LoadXRControllerModelSymbol],
+  )
+}
+
+export function getXRControllerComponentObject(
+  model: Object3D,
+  layout: XRControllerLayout,
+  componentId: XRControllerGamepadComponentId,
+) {
+  const component = layout.components[componentId]
+  // TODO: Add support for providing gamepad state
+  const firstVisualResponse = component.visualResponses[Object.keys(component.visualResponses)[0]]
+  if (!firstVisualResponse) return
+  const valueNode = model.getObjectByName(firstVisualResponse.valueNodeName)
+
+  return { object: valueNode }
+}
+
+const unboundControllerContext = createContext<XRControllerState | undefined>(undefined)
+
+export const UnboundController: FC<{ profileIds: string[] }> = ({ profileIds, children }) => {
```

Comment:
I wouldn't want to add a component for this use case since I dont think its that common
however, I'd appreciate a tutorial for this use case